File: LeetCode/2379. Minimum Recolors to Get K Consecutive Black Blocks/solution.cpp
/** 
 * Runtime: 4 ms
 * Memory Usage: 6.1 MB 
 */
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int i = 0, tmp = 0;
        while (i < k) {
            tmp += blocks[i++] == 'W';
        }
        int ans = tmp;
        for (; i < blocks.size() ; ++ i) {
            tmp += blocks[i] == 'W';
            tmp -= blocks[i - k] == 'W';
            ans = min(ans, tmp);
        }
        return ans;
    }
};

File: LeetCode/2379. Minimum Recolors to Get K Consecutive Black Blocks/solution.py
""" 
Runtime: 40 ms
Memory Usage: 14.8 MB
"""
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        ans, tmp = inf, 0
        for i, block in enumerate(blocks):
            if block == 'W':
                tmp += 1
            if i >= k and blocks[i - k] == 'W':
                tmp -= 1
            if i >= k - 1:
                ans = min(ans, tmp)
        return ans  
